### DISCUSSION TITLE
fast_forward插件的socks5代理支持udp的问题

### DISCUSSION THREAD

**Discussion by LittleGrocery** [IrineSistiana/mosdns#199] [Q&A]
目前根据wiki socks5代理并不支持用户名和密码认证，我主要是来问一下为什么不支持udp协议，只支持tcp协议呢？是不是实现有什么难题

**Comment by IrineSistiana** [comment]
实现麻烦。而且还不稳定，不常用。Dns over UDP 快成为历史了。

**Reply by LittleGrocery** [reply]
可是使用TCP查询DNS这样特征也太明显啦吧

**Reply by IrineSistiana** [reply]
用 udp 就不明显了么.....?

**Reply by LittleGrocery** [reply]
你好，我不了解深入的，但是觉得大多数dns请求都是基于udp的，本地到代理服务器过程当然没问题，但是在代理服务器到dns服务器这一段，看来有人使用tcp请求特征应该会十分明显吧

**Reply by IrineSistiana** [reply]
特征这玩意....争论起来都是玄学。能用就行。
